aiosmtpd/tests/test_handlers.py

```diff
@@ -186,6 +188,65 @@ def test_message_decoded(self):
         self.assertEqual(self.handled_message['X-RcptTos'], 'bart@example.com')
 
 
+class TestAsyncMessage(unittest.TestCase):
+    def setUp(self):
+        self.handled_message = None
+
+        class MessageHandler(AsyncMessage):
+            @asyncio.coroutine
+            def handle_message(handler_self, message, loop):
+                self.handled_message = message
+
+        self.handler = MessageHandler()
```

Comment:
I just took code from TestMessage and rewrote for AsyncMessage. Your points is right but I have not goal refact all other tests, I've just want to add new a feature :-)